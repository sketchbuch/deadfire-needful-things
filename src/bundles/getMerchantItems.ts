import { GameData, gameItem, GameItem } from './gamedatabundle'

export const getMerchantItems = (gameData: GameData) => {
  return gameData.map((data): GameItem => {
    const { uuid, item } = data

    return {
      ...gameItem,
      ...(item ? item : {}),
      ItemID: uuid,
    }
  })
}
