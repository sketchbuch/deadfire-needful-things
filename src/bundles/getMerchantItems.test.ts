import { mockData } from '../tests/mocks'
import { getMerchantItems } from './getMerchantItems'

describe('getMerchantItems()', () => {
  test('Items have the correct shape', () => {
    const result = getMerchantItems(mockData)

    expect(result).toHaveLength(2)
    expect(JSON.stringify(result)).toMatchSnapshot()
  })

  test('Items have their IDs correctly set', () => {
    const result = getMerchantItems(mockData)

    expect(result[0].ItemID).toBe(mockData[0].uuid)
    expect(result[1].ItemID).toBe(mockData[1].uuid)
  })

  test('Item properties can be over-ridden', () => {
    const result = getMerchantItems(mockData)

    if (mockData[1].item) {
      expect(result[1].LootListID).toBe(mockData[1].item.LootListID)
    }
  })
})
