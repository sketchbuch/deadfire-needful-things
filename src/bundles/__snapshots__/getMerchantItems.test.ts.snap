// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getMerchantItems() Items have the correct shape 1`] = `"[{\\"Conditional\\":{\\"Components\\":[],\\"Operator\\":0},\\"ItemID\\":\\"66f77403-092a-46be-a8b5-a4b4cd36e637\\",\\"LockedVisible\\":\\"false\\",\\"LootListID\\":\\"00000000-0000-0000-0000-000000000000\\",\\"MaxCount\\":1,\\"MinCount\\":1,\\"OutputChance\\":1,\\"Weight\\":1},{\\"Conditional\\":{\\"Components\\":[],\\"Operator\\":0},\\"ItemID\\":\\"4871decb-3312-4d46-8cf4-5e15e452d6ef\\",\\"LockedVisible\\":\\"false\\",\\"LootListID\\":\\"11111111-2222-3333-4444-555555555555\\",\\"MaxCount\\":1,\\"MinCount\\":1,\\"OutputChance\\":1,\\"Weight\\":1}]"`;
